index.js

```diff
@@ -33,13 +33,13 @@ module.exports.parent = function(obj) {
     return void 0;
   }
   return documentParentsMap.get(obj);
-}
+};
 
-function attachVirtualsMiddleware(schema) {
+function attachVirtualsMiddleware(schema, options = {}) {
   return function(res) {
-    if (this._mongooseOptions.lean && this._mongooseOptions.lean.virtuals) {
-      let virtuals = this._mongooseOptions.lean.virtuals;
-
+    const statement = Object.keys(this._mongooseOptions).length ? this._mongooseOptions.lean && this._mongooseOptions.lean.virtuals || options.enabledByDefault : options.enabledByDefault;
```

Comment:
Why `Object.keys(this._mongooseOptions).length` instead of `this._mongooseOptions.lean` ? In general, this logic is pretty hard to read, and also doesn't work well because if `options.enabledByDefault` is true but `this._mongooseOptions.lean.virtuals` is false, `statement` will still end up as true.

Reply:
It throws an error since someone can have `enabledByDefault` true and then not use the mongoose options. Its throws an error because this._mongooseOptions is an empty object so attempting to access `virtual` on the path throws an error. Isn't the idea of the PR that uses don't need `this._mongooseOptions.lean.virtuals`?